app/models/medical_shift.rb

```diff
@@ -5,16 +5,32 @@ class MedicalShift < ApplicationRecord
 
   monetize :amount
 
-  belongs_to :hospital
   belongs_to :user
 
   scope :by_hospital, MedicalShifts::ByHospitalQuery
   scope :by_month, MedicalShifts::ByMonthQuery
   scope :by_payd, MedicalShifts::ByPaydQuery
 
   validates :workload, presence: true
-  validates :date, presence: true
+  validates :start_date, presence: true
+  validates :start_hour, presence: true
   validates :amount_cents, presence: true
 
   validates :amount_cents, numericality: { greater_than_or_equal_to: 0 }
+
+  before_create :parsed_shift
+  before_save :parsed_title # rubocop:disable Rails/ActiveRecordCallbacksOrder
+
+  def parsed_shift
+    daytime_start = Time.new(2000, 0o1, 0o1, 0o7, 0o0, 0o0, 0o0)
+    daytime_finish = Time.new(2000, 0o1, 0o1, 18, 59, 0o0, 0o0)
+
+    return self.shift = "Diurno" if start_hour.between?(daytime_start, daytime_finish)
+
+    self.shift = "Noturno"
```

Comment:
With the translation file organized, we can replace these strings and use i18n for this.

Reply:
Done! I also created a new local to medical_shift model `config/locales/models/medical_shifts.yml` and I added translation for shift. It has the same problem that the other locale file, because the locale default is en, but we can change and fix in a new issue.